Data/ByteString.hs

```diff
@@ -889,6 +891,27 @@ take n ps@(BS x l)
     | otherwise = BS x n
 {-# INLINE take #-}
 
+-- | /O(1)/ 'takeEnd' @n xs@ is equivalent to @'drop' ('length' xs - n) xs@.
+-- Takes 'n' elements from end of bytestring.
+--
+-- >>> takeEnd 3 $ pack (c2w <$> "abcdefg")
+-- "efg"
+-- >>> takeEnd 0 $ pack (c2w <$> "abcdefg")
+-- ""
+-- >>> takeEnd 4 $ pack (c2w <$> "abc")
+-- "abc"
+takeEnd :: Int -> ByteString -> ByteString
+takeEnd n ps@(BS _ len)
+    | n <= 0    = empty
+    | n >= len  = ps
+    | otherwise = drop' (len - n) ps
+    where
+      drop' i xs@(BS x l)
+        | i <= 0    = xs
+        | i >= l    = empty
+        | otherwise = BS (plusForeignPtr x i) (l-i)
```

Comment:
Sorry for confusion. What I meant is that checks in lines 910-911 are always false (because 0 < len - n < len, once 0 < n < len), so you can inline the contents of line 912 `BS (plusForeignPtr x i) (l-i)` instead of `drop' (len - n) ps` in line 907.

Reply:
ahhh hehe, thought that might be the case but was unsure, I'll make the amend. 👍

Reply:
@Bodigrim which version would you prefer?
```
takeEnd :: Int -> ByteString -> ByteString
takeEnd n ps@(BS x len)
  | (len - n) <= 0   = ps
  | (len - n) >= len = empty
  | otherwise        = BS (plusForeignPtr x (len - n)) (len - (len - n))
```
```
takeEnd :: Int -> ByteString -> ByteString
takeEnd n ps@(BS x len)
  | newL <= 0   = ps
  | newL >= len = empty
  | otherwise   = BS (plusForeignPtr x newL) (len - newL)
  where newL = len - n
```

Reply:
The first one, and you can simplify `len - (len - n)` to `n`.

Reply:
it's late here?!?! I'll try use that as an excuse 😅 
Will make amend now, thanks for being patient it's appreciated.